### PR TITLE
docs: enable"dropdown" with all documentation versions

### PR DESCRIPTION
**Pull request by paoloyx**
This PR introduces two important changes to:

* how we publish our main documentation: **a dropdown will be shown with all published version of our docs.**
* how we make release: a dedicate `RELEASE.md` document has been added that covers all the necessary steps to perform a release

We cover both points in details below.

# Documentation version dropdown
Docusaurus offers a [versioning](https://docusaurus.io/docs/versioning) of the documentation that you write using it. In a nutshell, our main documentation site will show a dropdown (like the one below) containing references to every tagged release, and to the current/unreleased state of the documentation (shown as `Develop` 🚧) 

image

Every time we merge a documentation-related feature branch into `main`, the `Develop` 🚧 version of the documentation is updated. For what it concerns tagged releases, please read on.

# How to make a release
A dedicated `RELEASE.md` document has been added with every detail covered. When performing a release, the process will ensure that **a new documentation version will be created following the tagged release semantic version.**

### CODE REVIEW

File: .github/workflows/prepare-docs-for-release.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Prepare Docs for next release
```
Comment (paoloyx) [inline .github/workflows/prepare-docs-for-release.yaml:1]: In this file we add the action that creates a new "documentation version", by issuing the `npm run docusaurus docs:version` command

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Release Process
```
Comment (paoloyx) [inline RELEASE.md:1]: This document contains the detailed release process we're going to use

File: docs/docs/architecture.md [hunk 3]
```diff
@@ -23,5 +23,5 @@ It covers all the implemented APIs, starting from model creation and ending with
 
 ## SDK
 
-To interact with API programmatically, a [_Python SDK_](/python-sdk) is provided.
```
Comment (paoloyx) [inline docs/docs/architecture.md:26]: This is necessary, when versioning is enabled links must be written by linking at the markdown file. See: https://docusaurus.io/docs/versioning#link-docs-by-file-paths

**Review by robbenti — APPROVED** [review]

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -6,9 +6,21 @@
```
Comment (paoloyx) [inline docs/docusaurus.config.js:6]: The docusaurus config has been updated to enable versioning

File: docs/versioned_docs/version-v0.8.0/index.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (paoloyx) [inline docs/versioned_docs/version-v0.8.0/index.md:1]: From now on, we've added all documentation version for every previous tagged release

**Review by maocorte — APPROVED** [review]

**Pull request merged**